### PR TITLE
Add logging example

### PR DESCRIPTION
**Pull request by franziskuskiefer**
A simple example of how to use logging.
This is only an example of how to do this. Please start adding logging information to new code paths. I'll file a follow up with some more logging for existing code.

The logging level is controlled by the `RUST_LOG` env variable.

cc #142

### CODE REVIEW

File: src/codec.rs [hunk 1]
```diff
@@ -20,8 +20,7 @@ impl From<Error> for CodecError {
     // TODO: tbd in #83, also this direction shouldn't be necessary.
     fn from(e: Error) -> CodecError {
         match e {
-            Error::DecodingError => CodecError::DecodingError,
-            Error::UnsupportedCiphersuite => CodecError::DecodingError,
+            _ => CodecError::DecodingError,
```
Comment (kkohbrok) [inline src/codec.rs:23]: Couldn't this also be an `EncodingError`?

**Review by kkohbrok — APPROVED** [review]

File: src/codec.rs [hunk 2]
```diff
@@ -20,8 +20,7 @@ impl From<Error> for CodecError {
...
         }
     }
```
Comment (raphaelrobert) [inline src/codec.rs:25]: clippy warning

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
One clippy warning, the rest looks good!

File: src/codec.rs (hunk 1 above)
Comment (franziskuskiefer) [reply src/codec.rs:23]: This direction really shouldn't exist.. I guess I'll start on #83 at some point ... this getting annoying.

**Review by raphaelrobert — APPROVED** [review]

**Comment by franziskuskiefer** [comment]
I fixed the rebase issue to make the compiler happy. But I also started working on #83 to finally get error handling in better shape. This is getting annoying.

**Review by raphaelrobert — APPROVED** [review]

**Pull request merged**